localstack/services/sns/provider.py

```diff
@@ -169,9 +181,66 @@ def publish_message(
             message_attributes,
         )
     )
+
     return message_id
 
 
+def message_to_endpoint(
+    target_arn,
+    message,
+    structure,
+):
+    sns_client = aws_stack.connect_to_service("sns")
+    app_name = target_arn.split("/")[-2]
+    platform_name = target_arn.split("/")[-3]
+
+    try:
+        endpoint_attributes = sns_client.get_endpoint_attributes(EndpointArn=target_arn)[
+            "Attributes"
+        ]
+    except botocore.exceptions.ClientError:
+        LOG.warning(f"Missing attributes for endpoint: {target_arn}")
+        return
+
+    try:
+        platform_apps = sns_client.list_platform_applications()["PlatformApplications"]
+        app = [x for x in platform_apps if app_name in x["PlatformApplicationArn"]][0]
+    except IndexError:
+        LOG.warning(f"Missing Platform Application for endpoint: {target_arn}")
+        return
```

Comment:
When the endpoint or platform doesn’t exist, I think we should return an error to the user calling `publish`, as for now it fails silently and we don’t know if the call was correct or not. We could raise a `NotFound` exception if we can’t find the endpoint. For the platform, I’m guessing the same exception?  Normally, if the endpoint exists, the platform should as well except for race conditions. 

See: all endpoints deleted if platform is deleted:
https://stackoverflow.com/questions/68733553/does-deletion-of-a-sns-platform-application-delete-all-registered-endpoint

Reply:
I moved this code as you suggested, and if the endpoint doesn't exist it raises the correct error. For the platform i guess it should not happen due to the explanation you sent, but it looks like moto doesn't delete the endpoints when the platform app is deleted, so I'll make the necessary changes there.